spec/index.md

```diff
@@ -5365,8 +5439,70 @@ S with
 
 ```
 
+#### Trimming canister logs
+
+Canister logs can be trimmed if the total length of their contents exceeds 4KiB.
+
+Conditions
+
+```html
+
+S.canister_logs[CanisterId] = Older_logs · Newer_logs
+SUM { |l.content| | l <- Older_logs } > 4KiB
+
+```
+
+State after
+
+```html
+
+S with
+    canister_logs[CanisterId] = Newer_logs
+
+```
+
+#### IC Management Canister: Canister logs (query call) {#ic-mgmt-canister-fetch-logs}
+
+This section specifies management canister query calls.
+They are calls to `/api/v2/canister/<effective_canister_id>/query`
+with CBOR body `Q` such that `Q.canister_id = ic_principal`.
+
+The management canister offers the method `fetch_canister_logs`
+that can be called as a query call and
+returns logs of a requested canister.
+
+Conditions
+
+```html
+
+Q.canister_id = ic_principal
+Q.method_name = 'fetch_canister_logs'
+Q.arg = candid(A)
+A.canister_id = effective_canister_id
+S[A.canister_id].canister_log_visibility = Controllers or Public
```

Comment:
```suggestion
S[A.canister_id].canister_log_visibility = Public or Q.sender in S[A.canister_id].controllers
```

Reply:
`canister_log_visibility` is an enum, it should not contain the actual list of controllers, that's why I provided `Controllers or Public`.

or am I missing something? why did you add `Q.sender in S[A.canister_id].controllers`?

Reply:
The conditions should state that
- either the log visibility is _public_
- or the log visibility is _private_ and the sender of the message is a controller of the canister
Does this make sense?

Reply:
Yes it does, thank you!